x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlConfigMigrator.java

```diff
@@ -136,6 +137,25 @@ public void migrateConfigsWithoutTasks(ClusterState clusterState, ActionListener
                 }
         );
 
+        if (firstTime.get()) {
```

Comment:
What is the life time of this class? This `firstTime` thing only works if this is called multiple times on the same instance, this on purpose?

Reply:
The class has the same life time as the node and only runs on the master node. If it was a single node cluster or the master node never changed then this check would be sufficient to ensure the doc was never written more than once. The master node does change however, in which case the new master will have `firstTime == true` and will execute the code in the if statement. Setting `OpType == Create` stops the document being overwritten by the new master